main.c

```diff
@@ -81,32 +96,35 @@ int main(void) {
         }
         txb_heartbeat();
     }
+
     return (EXIT_SUCCESS);
 }
 
 static void __interrupt() interrupt_handler() {
     if (PIR5) {
-        //Handle CAN
+        // Handle CAN
         can_handle_interrupt();
     }
 
-    // deal with incoming UART bytes
-    if (PIR3bits.U1RXIF) {
-        //Handle GPS Interrupt
-        if (U1ERRIR) {
-            //error
-            U1ERRIR = 0; //ignore all errors
+    // UART message
+    if (PIR3bits.U1RXIF == 1) {
+        recieved_first_message = 1;
+
+        if (U1ERRIRbits.FERIF) {
+            // UART frame error
         }
-        if (recieved_first_message == 0) {
-            //we received something from uart for the first time.
-            recieved_first_message = 1;
+
+        if (U1ERRIRbits.RXFOIF) {
+            // UART overflowed
+            U1ERRIRbits.RXFOIF = 0;
         }
-        uint8_t byte = U1RXB;
-        gps_handle_byte(byte);
+
+        gps_handle_byte(U1RXB);
+
+        PIR3bits.U1RXIF = 0;
     }
 
     // Timer0 has overflowed - update millis() function
-    // This happens approximately every 500us
     if (PIE3bits.TMR0IE == 1 && PIR3bits.TMR0IF == 1) {
```

Comment:
any reason to delete this?

Reply:
I probably deleted it accidently when trying to fix uart bugs. I'll add them back